lib/prompt.ml

```diff
@@ -1,22 +1,33 @@
-let ask f =
+type default = Yes | No
+
+let ask f ~default_answer =
+  let options = match default_answer with Yes -> " [Y/n]" | No -> " [y/N]" in
   App_log.question (fun l ->
-      f (fun ?header ?tags fmt -> l ?header ?tags (fmt ^^ " [Y/n]")))
+      f (fun ?header ?tags fmt ->
+          l ?header ?tags (fmt ^^ Scanf.format_from_string options "")))
 
-let confirm ~question ~yes =
-  let rec loop () =
-    ask question;
+let rec loop ~question ~default_answer ~yes =
+  ask question ~default_answer;
```

Comment:
As mentioned below, we shouldn't "ask" if `--yes` was passed and just go straight to the point so this should be moved to the else branch I think.

Reply:
Yeah, asking a question when auto-confirming does feel a bit off... The reason why I did that was to be able to E2E test the different behaviour of `tag -d` when providing a commit-ish, that corresponds to the provided tag, or providing a "wrong" commit-ish (the difference lies in the question that asks for confirmation and in the default answer to that question). Do you think I should test that part of `tag` by adding a (general) flag `--answer-default` or should I simply not test that part?